components/ChipTextField/examples/MDCChipTextField.m

```diff
@@ -0,0 +1,360 @@
+  CGRect textRect = [super textRectForBounds:bounds];
+  textRect.origin.x = CGRectGetMaxX(self.chipsContainerView.frame);
+  textRect.size.width = MAX(0, textRect.size.width - CGRectGetWidth(self.chipsContainerView.frame));
+  return textRect;
+}
+
+- (CGRect)editingRectForBounds:(CGRect)bounds {
+  CGRect editingRect = [super editingRectForBounds:bounds];
+  return editingRect;
+}
+
+- (void)deleteBackward {
+  NSInteger cursorPosition = [self offsetFromPosition:self.beginningOfDocument
+                                           toPosition:self.selectedTextRange.start];
+  if (cursorPosition == 0) {
+    [self respondToDeleteBackward];
+  }
+  [super deleteBackward];
+}
+
+- (BOOL)hasTextContent {
+  return self.text.length > 0 || self.chipViews.count > 0;
+}
+
+- (void)clearText {
+  self.text = @"";
+  for (NSInteger index = self.chipViews.count - 1; index >= 0; --index) {
+    MDCChipView *chipView = self.chipViews[index];
+    [self removeChip:chipView];
+  }
+}
+
+#pragma mark - Deletion
+
+- (void)deselectAllChipsExceptChip:(MDCChipView *)chip {
+  for (MDCChipView *otherChip in self.chipViews) {
+    if (chip != otherChip) {
+      otherChip.selected = NO;
+    }
+  }
+}
+
+- (void)selectLastChip {
+  MDCChipView *lastChip = self.chipViews.lastObject;
+  [self deselectAllChipsExceptChip:lastChip];
+  lastChip.selected = YES;
+  UIAccessibilityPostNotification(UIAccessibilityAnnouncementNotification,
+                                  [lastChip accessibilityLabel]);
+}
+
+- (void)deselectAllChips {
+  [self deselectAllChipsExceptChip:nil];
+}
+
+- (void)removeChip:(MDCChipView *)chip {
+  [self.chipViews removeObject:chip];
+  [self.chipsContainerView removeChipView:chip];
+  [self clearChipsContainerOffsetWithConstant:kTextToEnterPlaceholderLength];
+
+  [self invalidateIntrinsicContentSize];
+  [self setNeedsLayout];
+}
+
+- (void)removeSelectedChips {
+  NSMutableArray *chipsToRemove = [NSMutableArray array];
+  for (MDCChipView *chip in self.chipViews) {
+    if (chip.isSelected) {
+      [chipsToRemove addObject:chip];
+    }
+  }
+  for (MDCChipView *chip in chipsToRemove) {
+    [self removeChip:chip];
+  }
+}
+
+- (BOOL)isAnyChipSelected {
+  for (MDCChipView *chip in self.chipViews) {
+    if (chip.isSelected) {
+      return YES;
+    }
+  }
+  return NO;
+}
+
+- (void)respondToDeleteBackward {
+  if ([self isAnyChipSelected]) {
+    [self removeSelectedChips];
+    [self deselectAllChips];
+  } else {
+    [self selectLastChip];
+  }
+}
+#pragma mark Notification Listener Methods
+
+- (void)textFieldDidBeginEditingWithNotification:(NSNotification *)notification {
+  [self setupEditingRect];
+}
+
+- (void)textFieldDidEndEditingWithNotification:(NSNotification *)notification {
+  [self clearChipsContainerOffsetWithConstant:0.0f];
+  [self.chipsContainerView scrollToLeft];
+}
+
+#pragma mark - MDCChipTextFieldScrollViewDataSource
+
+- (NSInteger)numberOfChipsOnScrollView:(MDCChipTextFieldScrollView *)scrollView {
```

Comment:
Tip: May "in" be better than "on" in this case?

Reply:
Addressed in 0f433959bc04c76f0573eef55b8d2c880d5caea7.